src/core/abr/representation_estimator.ts

```diff
@@ -434,16 +452,25 @@ export default function RepresentationEstimator({
     scoreCalculator.addSample(representation, requestDuration, segmentDuration);
   }
 
+  const switches$ = streamEvents$.pipe(
+    filter((e) : e is IABRRepresentationChangeEvent => e.type === "representationChange"),
+    tap(({ value }) => switchesStore.add(value)),
+    ignoreElements());
+
   const metrics$ = streamEvents$.pipe(
     filter((e) : e is IABRMetricsEvent => e.type === "metrics"),
-    tap(({ value }) => onMetric(value)),
+    tap(({ value }) => {
+      onMetric(value);
+      L2A.onMetricAdded(value);
+    }),
     ignoreElements());
 
   const requests$ = streamEvents$.pipe(
     tap((evt) => {
       switch (evt.type) {
         case "requestBegin":
           requestsStore.add(evt.value);
+          L2A.onSegmentDownloaded(evt.value);
```

Comment:
Either it is more something like `onSegmentDownloadStart` either it should be put inside `requestEnd` instead no?

In any case I found this integration pretty clean :+1: 
Though I did not get when we are going to rely on it here (not yet linked to the rest?).